pkg/k8shandler/elasticsearch/update.go

```diff
@@ -0,0 +1,117 @@
+package elasticsearch
+
+import (
+	"reflect"
+
+	"github.com/openshift/elasticsearch-operator/pkg/utils/comparators"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/elasticsearch-operator/pkg/logger"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+func UpdatePodTemplateSpec(name string, actual, desired *v1.PodTemplateSpec) bool {
+
+	changed := false
+
+	// check the pod's nodeselector
+	if !comparators.AreStringMapsSame(actual.Spec.NodeSelector, desired.Spec.NodeSelector) {
+		logrus.Debugf("Resource '%s' has different nodeSelector than desired", name)
+		actual.Spec.NodeSelector = desired.Spec.NodeSelector
+		changed = true
+	}
+
+	// check the pod's tolerations
+	if !comparators.AreTolerationsSame(actual.Spec.Tolerations, desired.Spec.Tolerations) {
+		logrus.Debugf("Resource '%s' has different tolerations than desired", name)
+		actual.Spec.Tolerations = desired.Spec.Tolerations
+		changed = true
+	}
+
+	// Only Image and Resources (CPU & memory) differences trigger rolling restart
+	for index := 0; index < len(actual.Spec.Containers); index++ {
```

Comment:
so one thing we did [1], which would be a safe thing to do... is ensure the two container names match before we do comparisons

[1] https://github.com/openshift/elasticsearch-operator/pull/315/files#diff-cac35ba9d75f18b147994710840e2b31R394-R403